### PR TITLE
post: A Grid In A Grid

### PR DESCRIPTION
**Pull request by kotzendekrabbe**
Posting about a grid in a grid.
This post was already a draft on DevTo but I wanted to use the new process for this one

### CODE REVIEW

File: posts/feli/a-grid-in-a-grid.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+title: A Grid In A Grid
+published: false
+description: You want to nest grid in grid with the same definition of columns & gaps? Check out these three possible solutions.
+tags: 'test, dummy'
```
Comment (s2-jossch) [inline posts/feli/a-grid-in-a-grid.md:5]: Maybe we should update the tags to something like: css, grid, thingsivelearned? :)

File: posts/feli/a-grid-in-a-grid.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    display: grid;
+    grid-template-columns: repeat(16, 1fr);
+    grid-column-gap: 16px;
+}
+```
+
+Now we want to nest elements. For example we want to have cell children in our wide area *cell-10*. We don't want to loose the *my-grid* definitions. But how can we solve that?
+
+### Step 2: Nesting Without Loosing The Parent Definition 
+There are different solutions, for different problems:
```
Comment (s2-jossch) [inline posts/feli/a-grid-in-a-grid.md:38]: I think the comma here can be removed :)

File: posts/feli/a-grid-in-a-grid.md (hunk 1 above)
Comment (kotzendekrabbe) [reply posts/feli/a-grid-in-a-grid.md]: haha, yes just fixed it while you was writing this comment - fancy.

File: posts/feli/a-grid-in-a-grid.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+- grid in a grid
+
+We will start with having a wrapper which is ignored by CSS.
+
+####Ignoring An Element CSSwise
+We can tell CSS that a HTML Element is not existing for styling by using display.
+__display: contents__ elements doesn't create a box themselves. They are replaced by a pseudo box and their child boxes. If you like you could also say __display: contents__ is telling the browser that there is now DOM-Element 😉
+
+```css
+.this-is-not-the-element-your-are-looking-for {
```
Comment (s2-jossch) [inline posts/feli/a-grid-in-a-grid.md:50]: love this 😆

File: posts/feli/a-grid-in-a-grid.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    display: contents;
+}
+```
+
+You may ask, why the heck should I have a HTML element that I don't need? There could be a several of reason. One could be that a framework need this wrapper element and you can't avoid it.
```
Comment (s2-jossch) [inline posts/feli/a-grid-in-a-grid.md:55]: [...] There could be several reasons [...]

File: posts/feli/a-grid-in-a-grid.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    grid-template-columns: subgrid;
+}
+```
+
+Sadly __subgrid__ doesn't has a good [browser support](https://caniuse.com/css-subgrid) yet.
+
+But what if we want to support at least the current browsers?
+
+####A Grid In A Grid
+To support the current browser we need to define a grid which fits to the parent column element. We will use the *cell-10* column to test it. Therefore we put in a new element and create a new class which we call *my-inline-grid-10*. The new grid will get a template column definition with the same amount of columns that the parent column is wide. In our case 10.
```
Comment (s2-jossch) [inline posts/feli/a-grid-in-a-grid.md:72]: Did you mean browser**s** or just the current one? :)

File: posts/feli/a-grid-in-a-grid.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+###Conclusions
+As we already said: **CSS Grid. Is. Not. Easy.** and if you haven't understood it completely yet, that's totally fine. 
+I hope that this short how-to may helped you to learn **how to nest a CSS grid in a grid**.
+
+If you like visual examples, here is my CodePen that includes all examples from above:
+{% codepen https://codepen.io/kotzendekrabbe/pen/jOqjqzE %}
+
+*!important* always check the [browser support] (https://caniuse.com)
+
+You have another solution, questions or feedback? Would be awesome if you leave a comment 🖤
```
Comment (s2-jossch) [inline posts/feli/a-grid-in-a-grid.md:92]: I really love how specific this article addresses one concrete problem & solution but puts it into a bigger context. It's also super helpful to find so many resources listed - great idea to also mention caniuse etc :)

Comment (kotzendekrabbe) [reply posts/feli/a-grid-in-a-grid.md:92]: thanks a bunch :) means a lot to me 💖

**Review by s2-jossch — APPROVED** [review]

**Pull request merged**